### PR TITLE
Fix test farm tests by using a local Pebble instance

### PR DESCRIPTION
**Pull request by bmw**
[As discussed in Mattermost](https://opensource.eff.org/eff-open-source/pl/yhtp4qu4zpfczm5wxmzxhndrto), our Apache test farm tests are failing because the CA certificate in the old version of boulder we have pinned expired over the weekend. This PR fixes that by running a local Pebble instance instead of an external boulder instance.

Probably the most controversial change in this PR is I removed our Apache test farm tests on ARM systems. I personally think this is fine. Looking through `git` history, we initially added ARM tests just because we had a lot of users on the architecture and we included running `test_apache2.sh` on ARM as part of that just because we could. See https://github.com/certbot/certbot/pull/7189.

I think the tests in `test_apache2.sh` are essentially architecture independent though and we're still testing on ARM in [test-farm-sdists](https://github.com/certbot/certbot/blob/421e8b6270aa869f5a6ae91c3504678ad3688bd3/tox.ini#L286-L288) where an ARM instance is defined in `targets.yaml` [here](https://github.com/certbot/certbot/blob/421e8b6270aa869f5a6ae91c3504678ad3688bd3/tests/letstest/targets.yaml#L28-L33). I personally think this is sufficient and not a big enough problem to block this PR, but let me know if you disagree!

You can see all tests passing with these changes at https://dev.azure.com/certbot/certbot/_build/results?buildId=3224&view=results.

### CODE REVIEW

**Review by adferrand — CHANGES_REQUESTED** [review]
The global feature seems fine to me, and the limitation to x86 only is not a big deal. Ideally we could open an issue to reenable these farm tests once Pebble exposes binaries for ARM architectures. It does not seem unachievable to propose that given foreign compilation is straight forward in Go.

However the way the HTTP-01 port is customized does not seem to fit well in the global picture here, having a partial support for non obvious reasons and so seems specifically designed for the edge case of the farm tests.

I propose an alternative: why not continuing the approach proposed in `certbot_call`, and make the `HTTP_01_PORT` Python constant configurable with an environment variable, then set this variable during the farm test ?

**Comment by bmw** [comment]
Thanks for the review.

If I'm understanding you correctly, there are two things you don't like:

1. Changing the http-01 port used by the server currently only works for Pebble without an HTTP proxy.
2. I used a flag rather than an environment variable to set the port that should be used.

For (1), I see no reason the other cases couldn't be supported, however, since we have no need for it right now, I didn't spend time writing and testing it. I could change the error message or add code comments about this to make why other cases aren't supported clearer if you like, but I personally prefer just writing the features we need instead of adding dead code that we may never use or may not even work anymore when we go to use it after it being untested for a long time.

For (2), I greatly prefer flags to environment variables. Beyond just personally thinking `run_acme_server --http-01-port 80` is a nicer and more conventional way to set options than `HTTP_01_PORT=80 run_acme_server`, problems with environment variables can be silently ignored like we saw at https://github.com/certbot/certbot/pull/8443#discussion_r541094899 while problems with flags should not so I think using flags is more robust.

I'm happy to make changes here, but I want to push back a bit on either making this work for all cases or removing a flag in favor of an environment variable.

Finally, today is the last day EFF is working this year and I would like to get tests on `master` fixed before I leave for vacation if possible. If you're able to work with me to quickly get a fix for these tests into a mergeable state, I'd really appreciate it!

**Comment by adferrand** [comment]
For the sake of fixing the tests before everyone leaves, I am willing to validate the fix, but in my opinion it is a quick and dirty approach that should be refactored as soon as possible, because it is lowering the overall quality of the code in certbot-ci.

I have no interest in making this approach work for the general case of integration tests, I simply think that using environment variables to configure objects in the `constant`module would actually reduce the number of line of code required.

`run_acme_server` has been primarily designed for the integration tests and to be used in conjonction with `certbot_test`, and using environment variables is quite coherent with the second case, since it is already used by `certbot_test` actually.

Here we would add a flag for a specific use case, out of the existing ones. Usually adding new flags for specific cases is refused in certbot.

**Review by adferrand — APPROVED** [review]

**Comment by adferrand** [comment]
As said, validated. Would it be possible to have an issue for a potential refractor, with a "significant" priority to secure in the middle term to study and maybe improve the approach?

File: certbot-ci/certbot_integration_tests/utils/acme_server.py [hunk 1]
```diff
@@ -223,7 +232,7 @@ def _prepare_http_proxy(self):
         print('=> Configuring the HTTP proxy...')
         mapping = {r'.+\.{0}\.wtf'.format(node): 'http://127.0.0.1:{0}'.format(port)
                    for node, port in self.acme_xdist['http_port'].items()}
-        command = [sys.executable, proxy.__file__, str(HTTP_01_PORT), json.dumps(mapping)]
+        command = [sys.executable, proxy.__file__, str(DEFAULT_HTTP_01_PORT), json.dumps(mapping)]
```
Comment (adferrand) [inline certbot-ci/certbot_integration_tests/utils/acme_server.py:235]: If you pass the user-defined http-01 port here, I think it will make it compatible with the proxy mode.

**Comment by bmw** [comment]
As discussed above and out-of-band, I'm going to merge this PR to get tests passing, but I'll open up an issue to discuss/track followup work shortly!
